ocaml/xapi/workload_balancing.ml

```diff
@@ -424,6 +443,9 @@ let retrieve_vm_recommendations ~__context ~vm =
       | Db_exn.Too_many_values (_,_,_) ->
         raise_malformed_response' "VMGetRecommendations"
           "Invalid VM or host UUID" "unknown"
+      | Float_of_string_failure ->
```

Comment:
Seems like you've deleted the code that raises this  exception, can the exception and this match be deleted too?

Reply:
Oops. The original exception can still be thrown.

Although it's unfortunate Stdlib raises a parametrized `Failure`, I think it's not worth having a special case to suppress the warning for matching against a string.

Reply:
Checking again, it's not even needed as the `val_num` function already does this. I'm changing its name so it's obvious what is doing and removing the match case.